solutions/devsprint-bruno-faganello-2/SpaceApp/Screens/HomeView.swift

```diff
@@ -10,23 +10,33 @@ import SwiftUI
 struct HomeView: View {
 
     @State var launches: [Launch] = []
+    private let service: SpaceXAPIProtocol
+    
+    init(service: SpaceXAPIProtocol) {
+        self.service = service
+    }
     
     var body: some View {
         NavigationView {
             List(launches) { launch in
                 Text("\(launch.name)")
             }
-            .onAppear() {
-                SpaceXAPI().fetchAllLaunches { launches in
-                    self.launches = launches ?? []
-                }
-            }.navigationTitle("Space App 🚀")
+            .onAppear(perform: fetchAllLaunches)
+            .navigationTitle("Space App 🚀")
+        }
+    }
+    
+    private func fetchAllLaunches() {
```

Comment:
[REQUEST CHANGED]
Let's create a ViewModel and Remove this fetch from View

Reply:
For this task? I will do this in the next Task, because I will ask if you what architecture you want to use kk (and this task only don't requested). but if you want I can do it.

Reply:
done! service was removed from HomView